docs/template_plugin/tests/functional/op_reference/logical.hpp

```diff
@@ -18,44 +18,49 @@ namespace LogicalOpsRefTestDefinitions {
 
 struct RefLogicalParams {
     ngraph::helpers::LogicalTypes opType;
-    Tensor input1;
-    Tensor input2;
+    std::vector<Tensor> inputs;
     Tensor expected;
 };
 
 struct Builder : ParamsBuilder<RefLogicalParams> {
     REFERENCE_TESTS_ADD_SET_PARAM(Builder, opType);
-    REFERENCE_TESTS_ADD_SET_PARAM(Builder, input1);
-    REFERENCE_TESTS_ADD_SET_PARAM(Builder, input2);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, inputs);
     REFERENCE_TESTS_ADD_SET_PARAM(Builder, expected);
 };
 
 class ReferenceLogicalLayerTest : public testing::TestWithParam<RefLogicalParams>, public CommonReferenceTest {
 public:
     void SetUp() override {
         const auto& params = GetParam();
-        function = CreateFunction(params.opType, params.input1.shape, params.input2.shape, params.input1.type);
-        inputData = {params.input1.data, params.input2.data};
+        function = CreateFunction(params.opType, params.inputs, params.inputs[0].type);
+        for (auto& input : params.inputs) {
+            inputData.push_back(input.data);
+        }
         refOutData = {params.expected.data};
     }
     static std::string getTestCaseName(const testing::TestParamInfo<RefLogicalParams>& obj) {
         const auto& param = obj.param;
         std::ostringstream result;
         result << "LogicalType=" << param.opType << "_";
-        result << "inpt_shape1=" << param.input1.shape << "_";
-        result << "inpt_shape2=" << param.input2.shape << "_";
-        result << "iType=" << param.input1.type << "_";
+        result << "inpt_shape1=" << param.inputs[0].shape << "_";
+        if (param.opType != ngraph::helpers::LogicalTypes::LOGICAL_NOT) {
+            result << "inpt_shape2=" << param.inputs[1].shape << "_";
+        }
```

Comment:
Looks like you can change this if to cycle

Reply:
Changed.